File: cpp/euler002.cpp
#include <iostream>

int main() {
    int buf1 = 1;
    int buf2 = 1;
    int next_fib_num = buf1 + buf2;
    int sum = 0;

    // Summing all even fibonacci numbers less than 4 million.
    while (next_fib_num < 4000000) {
        if (next_fib_num % 2 == 0) {
            sum += next_fib_num;
        }
        // Increment the correct buffer to get the next fibonacci number
        if (buf1 < buf2) { buf1 = next_fib_num; }
        else { buf2 = next_fib_num; }

        next_fib_num = buf1 + buf2;
    }

    // Print result
    std::cout << sum << std::endl;

    return 0;
}

File: python/Euler002.py
numberCap = 4000000

fibonacciNumbers = [1, 2] # Will contain ALL fibonacci sequence numbers
evenFibonacciNumbers = [] # Will only contain even fibonacci sequence numbers

lastItemInFibonacciSequence = 1

while fibonacciNumbers[lastItemInFibonacciSequence] < numberCap:
    lastItemInFibonacciSequence = len(fibonacciNumbers) - 1
    secondLastItemInFibonacciSequence = len(fibonacciNumbers) - 2

    fibonacciNumbers.append(fibonacciNumbers[lastItemInFibonacciSequence] + \
                            fibonacciNumbers[secondLastItemInFibonacciSequence])

fibonacciNumbers.pop() # Two values will exceed the numberCap, because there
fibonacciNumbers.pop() # are two items to start. Get rid of the unneeded values

print fibonacciNumbers[-2]

for i in fibonacciNumbers: # Weeding out any number that isn't even
    if i % 2 == 0:
        evenFibonacciNumbers.append(i)

print sum(evenFibonacciNumbers) 
